dask_cloudprovider/providers/generic/vmcluster.py

```diff
@@ -0,0 +1,135 @@
+            await asyncio.sleep(0.1)
+        self.cluster._log("Scheduler is running")
+
+    async def start(self):
+        """Create a VM."""
+        await super().start()
+
+    async def close(self):
+        """Destroy a VM."""
+        await super().close()
+
+
+class SchedulerMixin(object):
+    """A mixin for Schedulers.
+
+    """
+
+    def init_scheduler(self,):
+        self.name = "dask-scheduler"
+        self.command = "dask-scheduler --idle-timeout 300"
+
+    async def start_scheduler(self):
+        self.cluster._log("Creating scheduler instance")
+        ip = await self.create_vm()
+        self.address = f"tcp://{ip}:8786"
+        await self.wait_for_scheduler()
+
+
+class WorkerMixin(object):
+    """A Remote Dask Worker running on a VM.
+
+    """
+
+    def init_worker(self, scheduler: str, *args, worker_command=None, **kwargs):
+        self.scheduler = scheduler
+        self.worker_command = worker_command
+
+        self.name = f"dask-worker-{str(uuid.uuid4())[:8]}"
+        self.command = f"{self.worker_command} {self.scheduler}"
+
+    async def start_worker(self):
+        self.cluster._log("Creating worker instance")
+        self.address = await self.create_vm()
+
+
+class VMCluster(SpecCluster):
+    def __init__(self, n_workers=0, **kwargs):
+        self._n_workers = n_workers
+        self.scheduler_class = None
+        self.worker_class = None
+        self.scheduler_options = {}
+        self.worker_options = {}
+        self.docker_image = None
+        self.command = None
+        self.gpu_instance = None
+        self.bootstrap = None
+        self.auto_shutdown = True
+        super().__init__(**kwargs)
+
+    async def _start(self,):
+        while self.status == "starting":
+            await asyncio.sleep(0.01)
+        if self.status == "running":
+            return
+        if self.status == "closed":
+            raise ValueError("Cluster is closed")
+
+        self.scheduler_spec = {
+            "cls": self.scheduler_class,
+            "options": self.scheduler_options,
+        }
+        self.new_spec = {"cls": self.worker_class, "options": self.worker_options}
+        self.worker_spec = {i: self.new_spec for i in range(self._n_workers)}
+
+        with warn_on_duration(
+            "10s",
+            "Creating your cluster is taking a surprisingly long time. "
+            "This is likely due to pending resources. "
+            "Hang tight! ",
+        ):
+            await super()._start()
+
+    def render_cloud_init(self, *args, **kwargs):
+        loader = FileSystemLoader([os.path.dirname(os.path.abspath(__file__))])
+        environment = Environment(loader=loader)
+        template = environment.get_template("cloud-init.yaml.j2")
+        return template.render(**kwargs)
+
+    @classmethod
+    def get_cloud_init(cls, *args, **kwargs):
+        cluster = cls(*args, asynchronous=True, **kwargs)
+        cluster.auto_shutdown = False
```

Comment:
We don't want this hardcoded here, correct ?

Reply:
I think for the Packer use case it makes sense to default this to `False`, but perhaps now that it is configurable it should be configurable here too.